lib/friendly_shipping/services/tforce_freight/access_token.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'jwt'
+
+module FriendlyShipping
+  module Services
+    class TForceFreight
+      # Represents a JWT access token returned by TForce Freight. The access token can be
+      # used to make API requests. Once it expires, a new token must be created.
+      class AccessToken
+        # @return [String] the token's type
+        attr_reader :token_type
+
+        # @return [Integer] the token's expiration
+        attr_reader :expires_in
+
+        # @return [Integer] the token's extended expiration
+        attr_reader :ext_expires_in
+
+        # @return [String] the raw JWT access token
+        attr_reader :access_token
+
+        # @param token_type [String] the token's type (typically "Bearer")
+        # @param expires_in [Integer] the token's expiration
+        # @param ext_expires_in [Integer] the token's extended expiration (only applicable during a service outage)
+        # @see https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/675 Service Outage Resiliency Support
+        # @param access_token [String] the raw JWT access token
+        def initialize(token_type:, expires_in:, ext_expires_in:, access_token:)
+          @token_type = token_type
+          @expires_in = expires_in
+          @ext_expires_in = ext_expires_in
+          @access_token = access_token
+        end
+
+        # Decodes and returns the JWT access token.
+        # @return [Array<Hash>] the decoded access token
+        def decoded_access_token
+          @_decoded_access_token = JWT.decode(access_token, nil, false)
```

Comment:
Why in the world to they send an encoded jwt as the access token? That is really odd.

Reply:
Yeah, I have no idea. I think it's because they are using Azure Active Directory on the backend.

See https://developer.tforcefreight.com/resources/integration-end-user